ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -259,43 +259,59 @@ public void catchupStatusSuccessTest() throws InterruptedException, IOException
     assertTrue("This test needs more than one replica for the first partition to work", replicaIds.size() > 1);
 
     long acceptableLagInBytes = 100;
-
+    
     // cases with a given partition id
     // all replicas of given partition < acceptableLag
     generateLagOverrides(0, acceptableLagInBytes - 1);
-    doCatchupStatusTest(id, acceptableLagInBytes, ServerErrorCode.No_Error, true);
+    doCatchupStatusTest(id, acceptableLagInBytes, Short.MAX_VALUE, ServerErrorCode.No_Error, true);
     // all replicas of given partition = acceptableLag
     generateLagOverrides(acceptableLagInBytes, acceptableLagInBytes);
-    doCatchupStatusTest(id, acceptableLagInBytes, ServerErrorCode.No_Error, true);
+    doCatchupStatusTest(id, acceptableLagInBytes, Short.MAX_VALUE, ServerErrorCode.No_Error, true);
     // 1 replica of some other partition > acceptableLag
     String key = MockReplicationManager.getPartitionLagKey(otherPartRemoteRep.getPartitionId(),
         otherPartRemoteRep.getDataNodeId().getHostname(), otherPartRemoteRep.getReplicaPath());
     replicationManager.lagOverrides.put(key, acceptableLagInBytes + 1);
-    doCatchupStatusTest(id, acceptableLagInBytes, ServerErrorCode.No_Error, true);
+    doCatchupStatusTest(id, acceptableLagInBytes, Short.MAX_VALUE, ServerErrorCode.No_Error, true);
     // 1 replica of this partition > acceptableLag
     key = MockReplicationManager.getPartitionLagKey(id, thisPartRemoteRep.getDataNodeId().getHostname(),
         thisPartRemoteRep.getReplicaPath());
     replicationManager.lagOverrides.put(key, acceptableLagInBytes + 1);
-    doCatchupStatusTest(id, acceptableLagInBytes, ServerErrorCode.No_Error, false);
+    doCatchupStatusTest(id, acceptableLagInBytes, Short.MAX_VALUE, ServerErrorCode.No_Error, false);
+    // same result if num expected replicas == total count -1.
+    doCatchupStatusTest(id, acceptableLagInBytes, (short) (replicaIds.size() - 1), ServerErrorCode.No_Error, false);
+    // caught up if num expected replicas == total count - 2
+    doCatchupStatusTest(id, acceptableLagInBytes, (short) (replicaIds.size() - 2), ServerErrorCode.No_Error, true);
+    // caught up if num expected replicas == total count - 3
+    doCatchupStatusTest(id, acceptableLagInBytes, (short) (replicaIds.size() - 3), ServerErrorCode.No_Error, true);
     // all replicas of this partition > acceptableLag
     generateLagOverrides(acceptableLagInBytes + 1, acceptableLagInBytes + 1);
-    doCatchupStatusTest(id, acceptableLagInBytes, ServerErrorCode.No_Error, false);
+    doCatchupStatusTest(id, acceptableLagInBytes, Short.MAX_VALUE, ServerErrorCode.No_Error, false);
+    // if num expected replicas == 0, this succeeds
+    doCatchupStatusTest(id, acceptableLagInBytes, (short) 0, ServerErrorCode.No_Error, true);
```

Comment:
Why would someone set "0" as number of expected Replicas to have caught up.

Reply:
I was ambivalent about this since "0" doesn't affect the code in any way but I guess semantically it doesn't make sense to ask if "0 replicas have caught up". I turned 0 into a `Bad_Request`.